File: core/src/test/scala/tripaint/image/ImagePoolTest.scala
package tripaint.image

import munit.FunSuite
import tripaint.color.Color
import tripaint.image.ImagePool.SaveLocation
import tripaint.image.format.{SimpleStorageFormat, StorageFormat}

class ImagePoolTest extends FunSuite {
  val storageFormat: StorageFormat = SimpleStorageFormat

  test("locationOf should return None if the image doesn't exist") {
    val image = ImageStorage.fill(2, Color.Black)
    assertEquals(new ImagePool().locationOf(image), None)
  }

  test("locationOf should return the location of the image if it exists") {
    val image = ImageStorage.fill(2, Color.Black)
    val location = SaveLocation(null)
    val info = ImagePool.SaveInfo(null)

    val f = new ImagePool()
    f.set(image, location, info)

    assertEquals(f.locationOf(image), Some(location))
  }
}

File: core/src/test/kotlin/tripaint/image/ImagePoolTest.kt
package tripaint.image

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Nested
import tripaint.color.Color
import tripaint.image.format.SimpleStorageFormat
import tripaint.image.format.StorageFormat
import java.io.File
import kotlin.test.Test

class ImagePoolTest {
    val storageFormat: StorageFormat = SimpleStorageFormat

    @Nested
    inner class LocationOf {
        @Test
        fun `returns null if the image doesn't exist`() {
            val image = ImageStorage.fill(2, Color.Black)
            assertEquals(null, ImagePool().locationOf(image))
        }

        @Test
        fun `returns the location of the image if it exists`() {
            val image = ImageStorage.fill(2, Color.Black)
            val location = ImagePool.SaveLocation(File(""))
            val info = ImagePool.SaveInfo(storageFormat)

            val f = ImagePool()
            f.set(image, location, info)

            assertEquals(location, f.locationOf(image))
        }
    }
}